docs/designs/xref.md

```diff
@@ -4,6 +4,152 @@ Besides using file path to link to another file, DocFX also allows you to give a
+    > For the second last scenario, the output url would be `review.docs.microsoft.com` and the resolved uid url would be `docs.microsoft.com`, while clicking to this url, the user will go to another site `docs.microsoft.com` instead. Removing host for `uid` resolved URL can resolve this.
+    - The href of UID is from the same host name as the referencing repository.
+        - If the current branch is `live`, and the UID href is also from `live`, everything is OK
+        - If the current branch is `master`, then the output site is `review.docs`, but the resovled url is `docs` which is `live`, while browsing the UID href, the user should not jump to another site(`docs`)
+        ```yaml
+          # External UID `a` is defined in `docs`, whose title is `Title from docs`
+          inputs:
+            docs/b.md: Link to @a
+          outputs:
+            # The output site will be `review.docs`, which is non-live
+            docs/b.json: |
+                {"conceptual":"<p>Link to <a href=\"/a\">Title from docs</a></p>\n"}
+        ```
+        - If the current branch is `test`, and the build type is `PR build` or `branch build`, the UID href should fall back to `master` because `test` might not exist on UID definition repository.
+          ```yaml
+          # External UID `a` is defined in `docs`, whose title is `Title from docs`
+          inputs:
+            docs/b.md: Link to @a
+          outputs:
+            # The output site will be `review.docs`, which is non-live
+            docs/b.json: |
+                {"conceptual":"<p>Link to <a href=\"/a?branch=master\">Title from docs</a></p>\n"}
+          ``` 
+    - The href of UID is from a different host name as the referencing repository
+        - If the current branch is from a different site other than `docs`, we need to prioritize the uid from this site firstly if same uid defined in this site and `docs`
+          ```yaml
+          # External UID `a` is defined in `docs`, whose title is `Title from docs`
+          inputs:
+            azure-cn/a.md: |
+            ---
+            title: Title from azure-cn
+            uid: a
+            ---
+            azure-cn/b.md: Link to @a
+          outputs:
+            azure-cn/b.json: |
+                {"conceptual":"<p>Link to <a href=\"a\">Title from azure-cn</a></p>\n"}
+          ```
+- Reference to an external UID with versioning, the resolving logic should be the same as internal UID without versioning. We only take consideration of 1 version for now.
+- Reference to an external UID with multiple versionings (not support for now)
+    - The resolve logic should be smilar to the internal resolving
+- [UID defnition](#define-uid)
+- Needs to check if we have any conceptual repo output non-empty xrefmap.json. If yes, and auto-complete in xref service is still in use xref service needs to consume `xrefmap.json` instead of `xrefmap.yml` before v3 xref on-boarding.
```

Comment:
>Needs to check if we have any conceptual repo output non-empty xrefmap.json. If yes, and auto-complete in xref service is still in use xref service needs to consume `xrefmap.json` instead of `xrefmap.yml` before v3 xref on-boarding. [](start = 2, length = 232)

[https://github.com/aspnet/AspNetCore.Docs/search?q=uid&unscoped_q=uid](https://github.com/aspnet/AspNetCore.Docs/search?q=uid&unscoped_q=uid)